### ISSUE TITLE
[Feature Request]: add session variable management mechanism

### ISSUE THREAD

**Issue by daviszhen** [matrixorigin/matrixone#8080]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Is your feature request related to a problem?

```Markdown
There is no convenient mechanism to change the runtime behavior.
```

### Describe the feature you'd like

the mechanism:

- add session variable easily. (e.g. put the session variable def in cn.toml. or add session variable without reboot)
- the session variable can be carried down to all CNs running for this session/query.
- maybe, the session variable can be carried down to all DNs running for this session/query.

Write a design/tutorial on this.

**Comment by daviszhen** [comment]
@fengttt puts forward the idea firstly. I

**Comment by fengttt** [comment]
session variable does not necessarily need to go into conf.   asking user to issue a set statement is more or less ok.

but we want this, by default we want all these vairables under our own namespace.    make adding this kind of session variable easy.    for example, have one go file, if any dev need a new session variable like this, he/she should just add one line in the file and everything works.    maybe it is good to have all our session variable with prefix mo_xxx.    

make passing to multi cn work first.

**Comment by YANGGMM** [comment]
have been finished inhttps://github.com/matrixorigin/matrixone/pull/8517

**Comment by YANGGMM** [comment]
have been finished by @triump2020

**Comment by daviszhen** [comment]
完成

**Comment by aressu1985** [comment]
as dicussed by @daviszhen , the following item can not be implemented in 0.8:
the mechanism:

add session variable easily. (e.g. put the session variable def in cn.toml. or add session variable without reboot)
the session variable can be carried down to all CNs running for this session/query.
maybe, the session variable can be carried down to all DNs running for this session/query.
Write a design/tutorial on this.
